src/qemu/qemu_command.c

```diff
@@ -10937,11 +10937,21 @@ qemuBuildSerialChrDeviceProps(const virDomainDef *def,
         return NULL;
     }
 
-    if (virJSONValueObjectAdd(&props,
-                              "s:driver", virDomainChrSerialTargetModelTypeToString(serial->targetModel),
-                              "s:chardev", chardev,
-                              "s:id", serial->info.alias,
-                              NULL) < 0)
+    if (serial->targetModel == VIR_DOMAIN_CHR_SERIAL_TARGET_MODEL_ISA_SERIAL &&
+                    serial->target.port != -1) {
+            if (virJSONValueObjectCreate(&props,
```

Comment:
why it was changed to virJSONValueObjectCreate

Reply:
rebasing mistake.